### PR TITLE
Update to django 2.2

### PR DESCRIPTION
**Pull request by abrahammartin**
Now that the new LTS django 2.2 is out, update our current installation and all the dependencies to work with the new version of django.

This is deployed to tfc-app3.

### CODE REVIEW

**Comment by abrahammartin** [comment]
Can this PR get some love please? @jw35 @ijl20

**Comment by jw35** [comment]
Fall at the first fence! Did

    git checkout django2.2
    ./redeploy

It warned when doing 'pip instal' 

  django-haystack 2.6.1 has requirement Django<1.11, but you'll have django 2.2.7 which is incompatible

and then aborted in ./manage.py migrate with

    File "/home/tfc_prod/tfc_web_venv/lib/python3.5/site-packages/machina/apps/forum/admin.py", line 14, in <module>
        from django.core.urlresolvers import reverse
    ImportError: No module named 'django.core.urlresolvers'

Given that pip only said 

    Successfully installed Django-2.2.7 sqlparse-0.3.0

I expect that the problem is that nothing actually forced upgrades to haystack, machina, and probably others. I assume we need some `>=<version>` parameters in requirements.txt? Or should I be running `pip install -U -r requirements.txt` as a matter of course?

File: tfc_web/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,10 @@
-FROM python:3.6-slim-stretch
+FROM python:3.8.0-slim-buster
```
Comment (jw35) [inline tfc_web/Dockerfile]: Might want to revert this, given that we are going to be running 3.5 (tfc-app1-4) or 3.6 (tfc-app5) for the foreseeable future?

**Comment by jw35** [comment]
Needs 
    
    'DEFAULT_SCHEMA_CLASS': 'rest_framework.schemas.coreapi.AutoSchema'

adding to the `REST_FRAMEWORK` config item to keep CoreAPI working for the time being. See 5f00f27 in the django2.2-jw35-fixes branch for more detail.

**Comment by jw35** [comment]
> It warned when doing 'pip instal'
> 
> ```
> django-haystack 2.6.1 has requirement Django<1.11, but you'll have django 2.2.7 which is incompatible
> ```
> 
> and then aborted in ./manage.py migrate with
> 
> ```
> File "/home/tfc_prod/tfc_web_venv/lib/python3.5/site-packages/machina/apps/forum/admin.py", line 14, in <module>
>     from django.core.urlresolvers import reverse
> ImportError: No module named 'django.core.urlresolvers'
> ```

Resolved by removing and recreating the venv, activating it and then

    pip install --upgrade pip
    pip install -r tfc_web/tfc_web/requirements.txt

which will have installed current versions of all dependancies. I think running pip with -U would have done the same thing, except it would have left anything hand-installed in the past.

**Comment by jw35** [comment]
It looks as if this branch has a missing migration:

````
/manage.py makemigrations --dry-run
Migrations for 'csn':
  csn/migrations/0005_auto_20191129_1153.py
    - Delete model Destination
    - Delete model SensorData
````

Though it's probably not crucial.

**Comment by jw35** [comment]
Brief tests suggests this also works OK with the 'Download API' PR #345 added in.

**Comment by abrahammartin** [comment]
> Needs
> 
> ```
> 'DEFAULT_SCHEMA_CLASS': 'rest_framework.schemas.coreapi.AutoSchema'
> ```
> 
> adding to the `REST_FRAMEWORK` config item to keep CoreAPI working for the time being. See [5f00f27](https://github.com/SmartCambridge/tfc_web/commit/5f00f273fbf179471fb2dff356de3176c68ce96b) in the django2.2-jw35-fixes branch for more detail.

Thanks, cherry picked

**Comment by abrahammartin** [comment]
> Or should I be running `pip install -U -r requirements.txt` as a matter of course?

This is now in https://github.com/SmartCambridge/tfc_web/blob/master/redeploy.sh

**Comment by abrahammartin** [comment]
All changes done

**Comment by jw35** [comment]
> > Or should I be running `pip install -U -r requirements.txt` as a matter of course?
> 
> This is now in https://github.com/SmartCambridge/tfc_web/blob/master/redeploy.sh

That's quite a brave solution, since it's going to pull in updated copies of almost everything on every redeploy, but as long as you are happy...

**Pull request merged**